File: onderwijs4u-back-end/src/main/java/nl/han/devops/dto/UserSignupDTO.java
package nl.han.devops.dto;

public class UserSignupDTO {

    private String email;

    private String password;

    UserSignupDTO(String username, String password) {
        this.email = username;
        this.password = password;
    }

    public String getEmail() {
        return email;
    }

    public String getPassword() {
        return password;
    }
}

File: back-end/src/main/kotlin/nl/lpp/uncenshow/dto/user/UserSignupDTO.kt
package nl.lpp.uncenshow.dto.user

class UserSignupDTO(
    var username: String?,
    private var password: String?,
    var image: String?
) {

    constructor() : this(null, null, null)

    fun hashPassword(): String {
        return password!!.hashCode().toString()
    }

    fun isValidPassword(): Boolean {
        val regexString = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[@\$!%*?&])[A-Za-z\\d@$!%*?&]{8,50}\$"
        val regex = Regex(regexString)

        return regex.matches(this.password!!)
    }

    fun isValidUsername(): Boolean {
        return this.username!!.length >= 4 && this.username!!.length <= 50
    }
}
